Sources/Apodini/Request/DynamicProperty.swift

```diff
@@ -0,0 +1,214 @@
+//
```

Comment:
While I see the point of marking a type as discoverable by Apodini I would suggest to focus this implementation on what we currently name `RequestInjectable` and extend the functionality of `RequestInjectable` with the functionality found in `execute` and `apply`. Maybe even a more general term like `Injectable` might be more suiting as we might sometimes also inject information from a connection state (`@Environment(\.connection)`) or from the global environment. I think this would also solve some of the comments and discussions below about constraining `Dynamics` or `Injectables` by a specific protocol: https://github.com/Apodini/Apodini/pull/56#discussion_r543137013

Reply:
I heavily disagree. `RequestInjectable` is by far not the only use-case for `apply`. I use different protocols within my implementation of the WebSocket exporter and also the shared semantic model builder will use the execute function to find all `@Parameter`s at startup (it currently uses `Mirror` without support for `DynamicProperty` or `Dynamics`).

There will be way more use-cases (that do not fit a `Injectable` protocol) once we start working on `ObservableObject` and similar topics. My goal here was to provide abstracted **infrastructure-code** that leverages most of the work from working with protocols like `RequestInjectable`. As you said, this PR is all about enabling the user to **mark a type as discoverable by Apodini** and help the Apodini-internal developers access those types.

As of right now I don't see what the advantage of an `Injectable` protocol would be. Please help me see it. From my point of view `Injectable` does **not** cover all use-cases but the user of Apodini **should be able to** combine different use-cases (e.g. combine `@State` and `@Parameter` or `@Parameter` and `@ObservableObject`).